examples/defaulter-gen/generators/defaulter.go

```diff
@@ -432,19 +433,26 @@ func buildCallTreeForType(t *types.Type, root bool, existingDefaulters, newDefau
 	// base has been added already, now add any additional defaulters defined for this object
 	parent.call = append(parent.call, defaults.additional...)
 
+	// if the type already exists, don't build the tree for that
+	if existingTypes[t.Name.Name] {
+		return parent
+	}
+	// if type doesn't exist, mark it as existing
+	existingTypes[t.Name.Name] = true
```

Comment:
shouldn't we check that `Name` is actually files, i.e. not an anonymous type? Otherwise, we stop too early for those, don't we?